staging/src/k8s.io/api/apidiscovery/v2beta1/zz_generated.prerelease-lifecycle.go

```diff
@@ -0,0 +1,58 @@
+//go:build !ignore_autogenerated
+// +build !ignore_autogenerated
+
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by prerelease-lifecycle-gen. DO NOT EDIT.
+
+package v2beta1
+
+// APILifecycleIntroduced is an autogenerated function, returning the release in which the API struct was introduced as int versions of major and minor for comparison.
+// It is controlled by "k8s:prerelease-lifecycle-gen:introduced" tags in types.go.
+func (in *APIGroupDiscovery) APILifecycleIntroduced() (major, minor int) {
+	return 1, 26
+}
+
+// APILifecycleDeprecated is an autogenerated function, returning the release in which the API struct was or will be deprecated as int versions of major and minor for comparison.
+// It is controlled by "k8s:prerelease-lifecycle-gen:deprecated" tags in types.go or  "k8s:prerelease-lifecycle-gen:introduced" plus three minor.
+func (in *APIGroupDiscovery) APILifecycleDeprecated() (major, minor int) {
+	return 1, 29
+}
+
+// APILifecycleRemoved is an autogenerated function, returning the release in which the API is no longer served as int versions of major and minor for comparison.
+// It is controlled by "k8s:prerelease-lifecycle-gen:removed" tags in types.go or  "k8s:prerelease-lifecycle-gen:deprecated" plus three minor.
+func (in *APIGroupDiscovery) APILifecycleRemoved() (major, minor int) {
+	return 1, 32
+}
```

Comment:
these are fun for this API ... @deads2k, my understanding is that these serializations would *not* be deprecated and removed? if so, should these be set to 1.999 or something?

Reply:
My understanding is that the beta API will eventually be deprecated for a GA v2 API. Is that not the case?

Reply:
mm... deprecated, possibly, but I didn't think removed

Reply:
I'm open to a longer removal cycle since this API cannot tell clients to use discovery to determine if it's available, but I would prefer 6 or even 9 releases as opposed to forever.  Having a version allows for retiring the serialization.  If we didn't have a version, we could not retire it.

Reply:
> I'm open to a longer removal cycle since this API cannot tell clients to use discovery to determine if it's available

The  only drawback I see with removing the API is that kubectl/clients that only know about beta and non-aggregated would lose the benefits of aggregated if we remove beta (they wouldn't stop working, and it would be "transparent").

Reply:
I'm not sure I'd call performance of a 1.26 kubectl tanking against a new server "transparent" :-/

Reply:
I could live with dropping this once all supported kubectl / client-go versions have support for v2 (or v2beta2, etc). that'd be an overlap of probably 4 versions between availability of this and the next revision

Reply:
According to [version skew policy](https://kubernetes.io/releases/version-skew-policy/#kubectl), kubectl 1.26 is only promised to support up to kubernetes apiserver 1.27.

assuming we drop v2beta1 well after version 1.28, a  1.26 `kubectl` user affected by that should be well out of the supported version skew range.

granted, there might be other clients relying on v2beta1 serialization.

Reply:
Added deprecated for 1.32 and removed for 1.35 as a rough estimate (6 and 9 release cycles). Also added a comment indicating that these versions are estimates and may change in the future.